src/utils/cacheManagement.ts

```diff
@@ -0,0 +1,32 @@
+import { Client, ForumChannel, TextChannel, ThreadChannel } from "discord.js";
+
+export async function updateCache(client: Client) {
+    // Iterate through channels in the cache
+    client.channels.cache.forEach(async (channel) => {
+        try {
+            // If it's a ForumChannel
+            if (channel instanceof ForumChannel) {
+                // Fetch all its threads
+                const threads = await channel.threads.fetch()
+                // Iterate the threads
+                threads.threads.forEach(async (thread) => {
+                    // Ensure it's really a ThreadChannel, partially to satisfy TypeScript
+                    if (thread instanceof ThreadChannel && !thread.archived) {
+                        // Fetch the threads messages to the cache
+                        thread.messages.fetch({ limit: 100 })
+                    }
+                })
+            }
+            // If it's a TextChannel
+            if (channel instanceof TextChannel) {
+                // Fetch all messages to the cache
+                await channel.messages.fetch({ limit: 100 })
+            }
+        } catch (e) {
+            // We don't output the error because it tends to just be an API permission error that can be safely ignored
```

Comment:
Is there a way to check if it's that API permission error and if not then log it?

Reply:
It's not that it's not logged in, but that it doesn't have access. Kinda lazy error handling on my part. I'm updating it now to check specifically for the error code that corresponds with Missing Access, which afaik is just that the bot hasn't been given sufficient privileges to run the necessary commands on the channel, in this case it's moderators-only that I'm guessing it doesn't have permission to view messages in.